addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/handler/ChromecastHandler.java

```diff
@@ -141,24 +142,31 @@ public void dispose() {
 
     @Override
     public void initialize() {
-        final Object obj = getConfig().get(ChromecastBindingConstants.HOST);
-        if (!(obj instanceof String)) {
+        final Object ipAddress = getConfig().get(ChromecastBindingConstants.HOST);
+        if (!(ipAddress instanceof String)) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
                     "Cannot connect to Chromecast. IP address is invalid.");
             return;
         }
-        final String host = (String) obj;
+        final Object portnumber = getConfig().get(ChromecastBindingConstants.PORT);
+        if (!(portnumber instanceof Integer)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Cannot connect to Chromecast. port is invalid.");
+            return;
+        }
+        final String host = (String) ipAddress;
         if (StringUtils.isBlank(host)) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
                     "Cannot connect to Chromecast. IP address is not set.");
             return;
         }
+        final int port = (Integer) portnumber;
 
-        if (chromecast != null && !chromecast.getAddress().equals(host)) {
+        if (chromecast != null && !chromecast.getAddress().equals(host) && !(chromecast.getPort() == port)) {
```

Comment:
I think the right test would be: (chromecast != null && (!chromecast.getAddress().equals(host) || (chromecast.getPort() != port)))
We check etiehr if the IP is different or the port.